File: io_lib_errortest.rb
$dependencies << 'character_lib.o'
$dependencies << 'io_lib.o'
$dependencies << 'config_lib.o'
$compiler = ENV.fetch('MY_FORTRAN_DEBUG', 'gfortran -ffree-line-length-none -fmax-identifier-length=63 -pipe -cpp -C -Wall -fbounds-check -O0 -fbacktrace -ggdb -pg -DDEBUG')

setup <<-EOS
# include "../fortran_lib.h"
program runner
  USE_FORTRAN_LIB_H
  use, non_intrinsic:: io_lib

  implicit none

  Integer:: io
EOS

errortest 'Illegal form argument', "call mktemp(io, form = 'illegal')"

teardown <<-EOS
  stop
end program runner
EOS

File: io_lib_errortest.f90
# include "fortran_lib.h"
program main
   USE_FORTRAN_LIB_H
   use, intrinsic:: iso_fortran_env, only: input_unit, output_unit
   use, intrinsic:: iso_fortran_env, only: int32
   use, non_intrinsic:: io_lib

   implicit none

   Integer:: i


   read(input_unit, *) i


   select case(i)
   case(1)
      ! illegal argument for form
      block
         Integer(int32):: unit
         call mktemp(unit, form='illegal')
      end block
   case default
      write(output_unit, *) 1
   end select


   stop
end program main
